### PR TITLE
NEXUS-4851: upgrade to jetty8

### PR DESCRIPTION
**Pull request by mcculls**
Currently uses various testsuite and launcher snapshots that will eventually be staged before this is merged.

(also includes velo's UT fix so I can verify this on CI)

### CODE REVIEW

**Comment by mcculls** [comment]
First round of CI: https://builds.sonatype.org/job/nexus-oss-its-feature/238/ - linux ok, couple of failures on windows but they're intermittent failures that have occurred in the past with other builds

**Comment by cstamas** [comment]
:beer:

File: nexus/pom.xml [hunk 1]
```diff
@@ -1034,19 +1050,19 @@
       <dependency>
         <groupId>org.sonatype.http-testing-harness</groupId>
         <artifactId>server-provider</artifactId>
-        <version>0.5.0</version>
+        <version>0.5.1-SNAPSHOT</version>
```
Comment (peterlynch) [inline nexus/pom.xml]: not sure when the 'don't depend on external snapshots' rule is supposed to be applied...

**Comment by peterlynch** [comment]
+1 
~curious if we have tests that make sure nexus can boot with each of the jetty example config files.

File: nexus/pom.xml (hunk 1 above)
Comment (mcculls) [inline nexus/pom.xml]: ^ see above: "Currently uses various testsuite and launcher snapshots that will eventually be staged before this is merged"

**Comment by cstamas** [comment]
@peterlynch No, we don't have any, nor we had. Please create a issue to make tests validating all the "example" Jetty XML configurations we provide.

File: nexus/pom.xml (hunk 1 above)
Comment (peterlynch) [inline nexus/pom.xml]: yep my bad

**Comment by mcculls** [comment]
Waiting on merge of https://github.com/sonatype/sisu-jetty8/pull/1 and staged releases of sisu-jetty8 and appcontext-3.0

**Comment by mcculls** [comment]
CI results: +1 from https://builds.sonatype.org/job/nexus-oss-its-feature/249/ and https://builds.sonatype.org/job/nexus-oss-its-feature/256/

**Comment by cstamas** [comment]
@mcculls sisu-jetty8 pull merged, access rights granted, and staged both sisu-jetty8 and appcontext.

**Comment by cstamas** [comment]
Rebased branch against latest master.

**Comment by mcculls** [comment]
Thanks

**Pull request merged**